### DISCUSSION TITLE
Problem with Installation and setting up DIGIT

### DISCUSSION THREAD

**Discussion by madhan-vassar** [egovernments/Digit-Core#49] [Announcements]
Few questions i would like to address ::

1. What version of DIGIT are you using?
-> quickstart-v2.4

2. Are you attempting quickstart (single node deploy) or full installation of Digit?
-> quickstart

3. Is this installed using a laptop or AWS/GCP/Azure or State Data Centre?
-> laptop - DELL with processor -- 11th Gen Intel® Core™ i5-11320H @ 3.20GHz × 8 

4. Please provide link to the documentation used while installing.
-> https://core.digit.org/guides/installation-guide/quick-setup

5. Please provide specific step number where error was encountered.
-> 9th step/last step after proceeding with deployment

6. What is the OS? Provide version as well. For eg. Ubuntu 20.04
-> Ubuntu 20.04

7. Number of cores/vCPUs and RAM available?
-> No. of cores/vCPUs available - 8, Available RAM - 8.4 Gi

8. Free RAM currently available.(How to find free memory - )
-> Available RAM - 8.4 Gi

9. Description of the error along with a screenshot. Please paste the exact error encountered on the screen.
-> After proceeding with the deployment of DIGIT and answering all of the questions, the deployment has stopped displaying the following error message :
   "The connection to the server localhost:8080 was refused - did you specify the right host or port?", which I found after investigating that this type of error pops up when, it is not able to find the kubeconfig file but We have the config file of kubernetes and even provided it's path as answers to one of the questions asked while deploying DIGIT.

![Screenshot from 2022-11-09 11-23-01](https://user-images.githubusercontent.com/98875306/200762074-2c6b3c6b-3106-4146-84f6-6088065b1c93.png)

**Comment by subhashini-egov** [comment]
@gajendran-egov - can you please respond to this?

**Reply by subhashini-egov** [reply]
@madhan-vassar - if your laptop has 8.4 total RAM, 8.4 can't be free as well? Did you check the free memory?

**Reply by Karthik-egov** [reply]
@madhan-vassar 
1. Have you exported kube config path? 
If not use the below statement
export KUBECONFIG=path/to/kubeconfig
Note: you can find the kubeconfig in infra-as-code/terraform/sample-aws

2. Have you exported AWS profile?
If not use below statement
export AWS_PROFILE= profile-name

Follow the above steps.

**Comment by Karthik-egov** [comment]
@madhan-vassar

Have you exported kube config path?
If not use the below statement
export KUBECONFIG=path/to/kubeconfig
Note: you can find the kubeconfig in infra-as-code/terraform/sample-aws

Have you exported AWS profile?
If not use below statement
export AWS_PROFILE= profile-name

Follow the above steps

**Comment by upyog-niua** [comment]
Getting similar error. Kubeconfig path and AWS profile have been exported. How to resolve?

**Reply by ssaranga** [reply]
getting similar error while trying to deploy on aws.

Able to connect k3d cluster from local machine, but when executing sudo go run digit_setup.go received error. 
----  
saranga@MacBook-Pro egov-deployer % kubectl cluster-info
Kubernetes control plane is running at https://ec2-3-110-90-109.ap-south-1.compute.amazonaws.com:6550
CoreDNS is running at https://ec2-3-110-90-109.ap-south-1.compute.amazonaws.com:6550/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
Metrics-server is running at https://ec2-3-110-90-109.ap-south-1.compute.amazonaws.com:6550/api/v1/namespaces/kube-system/services/https:metrics-server:/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
----


**Error:**
exit status 1: The connection to the server 0.0.0.0:51092 was refused - did you specify the right host or port?

panic: exit status 1: The connection to the server 0.0.0.0:51092 was refused - did you specify the right host or port?


@madhan-vassar @jagankumar-egov @subhashini-egov

**Comment by nikhilmulinti** [comment]
Hi @ssaranga Please make sure kubeconfig is exported before you run the go script, you can verify by running kubectl get nodes, see if you are able to connect to cluster
